### PR TITLE
✨ server: add card limit update handler

### LINKED ISSUE #889: server: increase card limit after extra kyc approved

### PR DESCRIPTION
**Pull request by aguxez**
Closes #889 

## Summary by CodeRabbit

* **New Features**
  * Added a card-limit case: approved cases can trigger automatic card limit updates (amount applied in cents, weekly frequency) by resolving related inquiry references and updating active cards.
  * Added a new KYC scope "cardLimit" and inquiry flow to create or resume card-limit inquiries.

* **Bug Fixes / Reliability**
  * Clear response codes and graceful handling for missing data or lookup failures; errors are captured without breaking the webhook.

* **Tests**
  * Extensive tests covering card-limit flows, ignored inquiries, edge cases, and failure modes.

* **Chores**
  * Added patch release metadata.

---

<a href="https://app.devin.ai/review/exactly/exa/pull/898" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: server/hooks/persona.ts [hunk 1]
```diff
@@ -211,6 +241,37 @@ export default new Hono().post(
...
+      if (!credential) {
+        captureException(new Error("no credential"), { level: "error", contexts: { credential: { referenceId } } });
+        return c.json({ code: "no credential" }, 200);
+      }
+      if (!credential.pandaId) return c.json({ code: "no panda" }, 200);
+      if (card)
+        await updateCard({ id: card.id, limit: { amount: cardLimitUsd * 100, frequency: "per7DayPeriod" } }).catch(
+          captureException,
+        );
+      return c.json({ code: "ok" }, 200);
```
[the remark this answers is not in the document]
Comment (aguxez) [reply server/hooks/persona.ts]: I don't think this is a valid reason. Frozen cards should not be able to get their limits increased to start with so it's ok we simply ignore them. It's also not possible for a limit increase to be requested if you don't have a card already issued

File: server/api/kyc.ts [hunk 2]
```diff
@@ -141,6 +175,56 @@ export default new Hono()
       setUser({ id: parse(Address, credential.account) });
       setContext("exa", { credential });
 
+      if (scope === "cardLimit") {
+        if (await getPendingInquiryTemplate(credentialId, "basic")) return c.json({ code: "not started" }, 400);
```
[the remark this answers is not in the document]
Comment (aguxez) [reply server/api/kyc.ts]: this is on purpose, a person without an approved kyc could still have a pandaId so we need to ensure they actually finished kyc

File: server/api/kyc.ts [hunk 3]
```diff
@@ -173,15 +266,13 @@ export default new Hono()
           return c.json({ code: "failed", legacy: "kyc failed" }, 400);
         case "completed":
         case "needs_review":
-          return c.json({ code: "failed", legacy: "kyc failed" }, 400); // TODO send a different response
+          return c.json({ code: "processing", legacy: "kyc failed" }, 400);
```
[the remark this answers is not in the document]
Comment (aguxez) [reply server/api/kyc.ts:266]: This is expected

File: server/hooks/persona.ts [hunk 4]
```diff
@@ -210,7 +244,47 @@ export default new Hono().post(
...
+      if (credential?.pandaId && credential.cards[0]) {
+        await updateCard({
+          id: credential.cards[0].id,
+          limit: { amount: limitUsd * USD_TO_CENTS, frequency: "per7DayPeriod" },
+        });
+      }
+      await updateCardLimit(referenceId, limitUsd).catch((error: unknown) => {
+        captureException(error, {
+          level: "error",
+          contexts: {
```
[the remark this answers is not in the document]
Comment (aguxez) [reply server/hooks/persona.ts:268]: A missing account on Persona is not possible to trigger this webhook handler.

File: server/api/kyc.ts [hunk 5]
```diff
@@ -108,12 +153,12 @@ export default new Hono()
           return c.json({ code: "not started", legacy: "kyc not started" }, 400);
         case "completed":
         case "needs_review":
-          return c.json({ code: "bad kyc", legacy: "kyc not approved" }, 400); // TODO send a different response for this transitory statuses
+          return c.json({ code: "processing", legacy: "kyc not approved" }, 400);
```
[the remark this answers is not in the document]
Comment (aguxez) [reply server/api/kyc.ts:157]: this was done on purpose

File: server/hooks/persona.ts [hunk 6]
```diff
@@ -210,7 +245,58 @@ export default new Hono().post(
...
+      const credential = await database.query.credentials.findFirst({
+        columns: { pandaId: true },
+        where: eq(credentials.id, referenceId),
+        with: { cards: { columns: { id: true }, where: inArray(cards.status, ["ACTIVE", "FROZEN"]), limit: 1 } },
+      });
+      if (!credential) {
+        captureException(new Error("no credential"), { level: "error", contexts: { credential: { referenceId } } });
+        return c.json({ code: "ok" }, 200);
+      }
+      await updateCardLimit(referenceId, limitUsd).catch((error: unknown) => {
```
[the remark this answers is not in the document]
Comment (aguxez) [reply server/hooks/persona.ts:265]: No op
